File: exercises/java/src/main/java/cards/SuperDeck.java
package cards;

import java.util.ArrayList;
import java.util.Collections;

public class SuperDeck implements Deck {

    public AnimalDeck animalDeck = new AnimalDeck();
    public PlayingCardDeck playingCardDeck = new PlayingCardDeck();

    public ArrayList<Card> cards;

    public SuperDeck() {
        cards = new ArrayList<>();

        for (int i = 0; i < animalDeck.getCards().length; i++) {
            cards.add(animalDeck.deal());
        }

        for (int i = 0; i < playingCardDeck.getCards().length; i++) {
            cards.add(playingCardDeck.deal());
        }
    }

    @Override
    public String[] getCards() {
        String[] result = new String[cards.size()];
        for (int i = 0; i < cards.size(); i++) {
            Card card = cards.get(i);
            result[i] = card.toString();
        }
        return result;
    }

    @Override
    public Card deal()  {
        return cards.remove(0);
    }

    @Override
    public void shuffle() {
        Collections.shuffle(cards);
    }
}

File: exercises/swift/SnapGameOOP/SnapGameOOP/GameLogic/Cards/SuperDeck/SuperDeck.swift
//
//  SuperDeck.swift
//  SnapGameOOP
//
//  Created by Andrew Harris on 19/11/2019.
//

import Foundation

class SuperDeck {
    
    private var cards = [Card]()
    
    init(decks: [Deck]) {
        decks.forEach { (deck) in
            cards.append(contentsOf: deck.getCards())
        }
    }
    
    init() {
        let animalDeck = AnimalDeck()
        let playingDeck = PlayingDeck.generateNewDeck()
        let decks: [Deck] = [animalDeck, playingDeck]
        decks.forEach{ (deck) in
            cards.append(contentsOf: deck.getCards())
        }
    }
    
}

extension SuperDeck: Deck {
    
    func getCards() -> [Card] {
        return cards
    }
    
    func shuffle() {
        cards.shuffle()
    }
    
    func deal() -> Card {
        return cards.remove(at: 0)
    }
    
}
